firestore-counter/clients/ios/Sources/FirestoreCounter/FirestoreCounter.swift

```diff
@@ -0,0 +1,91 @@
+import FirebaseFirestore
+import Firebase
+
+public class FirestoreShardCounter {
+  private let firestore: Firestore
+  private let shardId = UUID().uuidString
+  private var shards = [String: Double]()
+  private let collectionId = "_counter_shards_"
+  private let shardRef: CollectionReference
+  private var docRef: DocumentReference
+  private var field: String
+
+  public init(docRef: DocumentReference, field: String) {
+    self.docRef = docRef
+    self.field = field
+    self.firestore = docRef.firestore
+    self.shardRef = docRef.collection(collectionId)
+    shards[self.docRef.path] = Double(0)
```

Comment:
No need to use `Double(0)` - you've already told the compiler about the type in line 7.
Also don't need to use `self` here.

Replace with:
```
shards[docRef.path] = 0
```